sdk/src/types/block/output/output_id_proof.rs

```diff
@@ -18,10 +22,41 @@ use crate::types::block::slot::SlotIndex;
 pub struct OutputIdProof {
     pub slot: SlotIndex,
     pub output_index: u16,
-    pub transaction_commitment: String,
+    #[cfg_attr(feature = "serde", serde(with = "prefix_hex_bytes"))]
+    pub transaction_commitment: [u8; 32],
     pub output_commitment_proof: OutputCommitmentProof,
 }
 
+impl OutputCommitmentProof {
+    pub fn new(output_ids: &[OutputId], index: u16) -> Self {
+        let n = output_ids.len();
+        debug_assert!(n > 0 && index < n as u16, "n={n}, index={index}");
+
+        // Handle the special case where the "value" makes up the whole Merkle Tree.
+        if n == 1 {
+            return OutputCommitmentProof::ValueHash(ValueHash::new(&output_ids[0]));
+        }
+
+        // Select a `pivot` element to split `data` into two slices `left` and `right`.
+        let pivot = largest_power_of_two(n as _) as u16;
+        let (left, right) = output_ids.split_at(pivot as _);
+
+        // Produces the Merkle hash of a sub tree not containing the `value`.
+        let subtree_hash = |output_ids: &[OutputId]| {
+            let values = output_ids.into_iter().map(OutputId::to_bytes).collect::<Vec<_>>();
```

Comment:
```
    Checking iota-sdk v1.1.4 (/home/runner/work/iota-sdk/iota-sdk/sdk)
error[E0412]: cannot find type `Vec` in this scope
  --> sdk/src/types/block/output/output_id_proof.rs:46:83
   |
46 |             let values = output_ids.into_iter().map(OutputId::to_bytes).collect::<Vec<_>>();
   |                                                                                   ^^^ not found in this scope
   |
help: consider importing one of these items
   |
4  + use alloc::vec::Vec;
   |
4  + use serde::__private::Vec;
```

Also can you add some test from the data here https://github.com/iotaledger/tips/blob/tip45/tips/TIP-0045/tip-0045.md#output-id-proof-1 please?

Reply:
Or open an issue so we add tests in another PR